extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/ServerSideEncryptingAmazonS3.java

```diff
@@ -119,4 +130,16 @@ public void deleteObject(String bucket, String key)
   {
     amazonS3.deleteObject(bucket, key);
   }
+
+  public boolean upload(PutObjectRequest request) throws java.lang.InterruptedException
+  {
+    if (transferManager == null) {
+      putObject(request);
```

Comment:
Does the `PutObjectResult` need to be checked for errors?

Reply:
> Does the `PutObjectResult` need to be checked for errors?

I checked through the whole code base, `PutObjectResult` was not checked for errors. The upper stream code checks for exceptions thrown by putObject().